SelfAndPeerEvaluations/partner_eval_quiz_creator.py

```diff
@@ -0,0 +1,161 @@
+        self.assignment_group_id = (
+            assignment_group_id
+            if assignment_group_id
+            else self.__find_partner_eval_ag_id(course)
+        )
+        self.unlock_date = unlock_date
+        self.due_date = due_date
+        self.lock_date = lock_date
+
+    def upload_to_canvas(self) -> None:
+        print("Uploading quiz assignment to canvas")
+        quiz_info = self.__create_quiz_info()
+        quiz_questions = self.__create_quiz_questions()
+
+        quiz = self.course.create_quiz(quiz_info)
+        for quiz_question in quiz_questions:
+            quiz.create_question(question=quiz_question)
+
+        canvas_assignment = course.get_assignment(quiz.assignment_id)
+        canvas_assignment.edit(
+            assignment={"omit_from_final_grade": True, "published": True}
+        )
+
+        print("Finished upload!")
+
+    def __create_quiz_info(self) -> dict:
+        return {
+            "title": self.assignment_name,
+            "description": self.json_questions["description"],
+            "quiz_type": self.json_questions["quiz_type"],
+            "assignment_group_id": self.assignment_group_id,
+            "scoring_policy": self.json_questions["scoring_policy"],
+            "published": self.json_questions["published"],
+            "show_correct_answers": self.json_questions["show_correct_answers"],
+            "unlock_at": self.unlock_date,
+            "due_at": self.due_date,
+            "lock_at": self.lock_date,
+            "allowed_attempts": self.json_questions["allowed_attempts"],
+        }
+
+    def __create_quiz_questions(self) -> list[dict]:
+        """
+        Iterates through all the questions in the JSON file's question list
+        Most of the questions in the JSON file are properly formated formated
+        for canvasapi, but some have missing fields (answer section, pts possible)
+        :return: list of questions ready to send to canvasapi's create_question()
+        """
+        quiz_questions = []
+        multiple_choice_answers = dict(
+            enumerate(
+                {"answer_text": choice, "answer_weight": 1}
+                for choice in [
+                    "Strongly disagree",
+                    "Disagree",
+                    "Agree",
+                    "Strongly agree",
+                ]
+            )
+        )
+        for question in self.json_questions["questions"]:
+            if question["question_name"] == "Partner Identification":
+                question["answers"] = self.__create_identify_partner_answers()
+            elif (
+                question["question_type"] == "multiple_choice_question"
+                and "answers" not in question
+            ):
+                question["answers"] = multiple_choice_answers
+            if "points_possible" not in question:
+                # by default, we want questions to be 1 point
+                question["points_possible"] = 1
```

Comment:
Well done allowing the overrides. If we switch to the default json I mentioned earlier this can be simplified to a loop that looks something like
```
for default_key, default_value in default_json:
  if default_key not in question:
     question[default_key] = default_value
```